manager/manager.go

```diff
@@ -808,3 +942,164 @@ func managerNode(nodeID string) *api.Node {
 		},
 	}
 }
+
+// raftDEKRotator manages the raft DEK keys using TLS headers
+type raftDEKRotator struct {
+	mu           sync.Mutex
+	currentKEK   []byte
+	nextDEK      []byte
+	rotationChan chan struct{}
+	kw           ca.KeyWriter
+}
+
+func (r *raftDEKRotator) GetCurrentKEK() []byte {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	return r.currentKEK
+}
+
+func (r *raftDEKRotator) SetKeyWriter(kw ca.KeyWriter) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.kw = kw
+}
+
+func (r *raftDEKRotator) GetNewKey() []byte {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	return r.nextDEK
+}
+
+func (r *raftDEKRotator) RotationNotify() chan struct{} {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	return r.rotationChan
+}
+
+func (r *raftDEKRotator) FinishKeyRotation(key []byte) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if bytes.Equal(key, r.nextDEK) {
+		r.nextDEK = nil
+	}
+	if r.kw != nil {
+		return r.kw.UpdateHeaders(func(headers map[string]string, kek []byte) error {
+			r.currentKEK = kek
+			// no matter what, set the current key
+			headerStr, err := encodePEMHeaderValue(key, kek)
+			if err != nil {
+				return err
+			}
+			headers[defaultRaftDEKKey] = headerStr
+
+			if pendingDEKStr, ok := headers[defaultRaftDekKeyPending]; ok {
+				pendingKey, err := decodePEMHeaderValue(pendingDEKStr, kek)
+				if err != nil {
+					return err
+				}
+				// that's weird, but ok - I guess rotate again
+				if !bytes.Equal(key, pendingKey) {
```

Comment:
`subtle.ConstantTimeCompare`? (as with all of the above comparisons, I'm not sure it actually matters, just seems like a good practice when comparing secret values)